### DISCUSSION TITLE
FFMPEG extension to handle video files (.mp4)

### DISCUSSION THREAD

**Discussion by dangerstudios** [directus/directus#19223] [Feature Requests]
### Summary

Ideally integrating something like ffmpeg to transcode a video and generate a thumbnail on upload of a video file. 

### Basic Example

1. Drag a video file into directus
2. Automatic transcode with some options like generating a thumbnail & save those options to a preset so you don't have to choose them every time you upload a video.
3. Populate any related meta data fields. 

### Motivation

Directus does not handle video very well. 

### Detailed Design

This feature would be used by anyone wanting to manage video files with directus. At the moment Directus treats a video file (specifically an .mp4 file) like an unknown file type. 

### Must Have:
- Transcoding 
- Thumbnail generation

### Should Have:
- metadata generation, height, width, duration, fps, etc. 
- options that can be saved to a preferences file. 

### Could Have:
- additional ffmpeg options. 

### Won't Have:
- anything that ffmpeg can't do.

### Drawbacks

increased installation size by adding ffmpeg.

### Alternatives

some ffmpeg alternative? I don't really see a better alternative at present. 

### Adoption Strategy

This could potentially be an extension? 

### Unresolved Questions

How feasible it would be to make an ffmpeg extension?

**Comment by rijkvanzanten** [comment]
When researching this in the past, the problem was the reliance on native binaries. In the present though, where the Docker image is the primary distribution method, that might no longer be a blocker 🙂

**Reply by dangerstudios** [reply]
Ah I see, There is a web assembly port of ffmpeg which looks pretty cool:
https://github.com/ffmpegwasm/ffmpeg.wasm

**Reply by rijkvanzanten** [reply]
Ohhh that's very interesting

**Reply by navanshu** [reply]
I believe a full solution would be better than maintaining and building an extension of this maybe if there is a service or another docker hostable software that would be easier to implement it with Directus, like minio or redis like an API

Something like
1. https://github.com/mediacms-io/mediacms
2. https://github.com/kaltura/nginx-vod-module
3. https://github.com/fastogt/fastocloud
4. https://github.com/ossrs/srs

**Comment by rijkvanzanten** [comment]
Heya! Thanks for this great feature request. There was actually another feature request opened about a month earlier that I think encompasses the same underlying feature needs, so lets merge the two together to make sure we don't lose track 🙂 

Lets continue the implementation discussion for this over on https://github.com/directus/directus/discussions/18947
